monitoring/check_collectd_mlab.py

```diff
@@ -0,0 +1,461 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Summary:
+  check_collectd_mlab.py is a nagios plugin that checks the health of
+  collectd-mlab.
+
+  check_collectd_mlab.py should be installed and run from the host context.
+
+When check_collectd_mlab.py returns an OKAY status, the following are true:
+ * collectd is installed in the utility slice.
+ * collectd responds to commands sent over the collectd unix socket.
+ * collectd is running on a writable filesystem.
+ * vsys configuration is correct for the backend, frontend, and slice acl.
+ * resource usage by the collectd process and slice are normal.
+ * TODO: make resource usage checks configurable.
+ * TODO: check the last disk sync time recently and without too much error.
+
+If any of the above checks fail, check_collectd_mlab.py exits with a nagios
+error code and descriptive message.
+
+Installation:
+  /usr/lib/nagios/plugins/
+
+Example usage:
+  $ ./check_collectd_mlab.py
+  OKAY: Blue skies smiling at me: 1.27 sec
+
+If collectd is not running:
+  $ ./check_collectd_mlab.py
+  CRITICAL: collectd unixsock not present:
+       /vservers/mlab_utility/var/run/collectd-unixsock.
+"""
+
+import os
+import signal
+import socket
+import subprocess
+import sys
+import time
+
+
+# These values expect that:
+# * This script runs in host context.
+# * collectd runs in mlab_utility slice.
+# * collectd uses default pidfile name.
+# * collectd uses default unixsock name.
+SLICENAME = 'mlab_utility'
+COLLECTD_BIN = '/vservers/%s/usr/sbin/collectd' % SLICENAME
+COLLECTD_NAGIOS = '/vservers/%s/usr/bin/collectd-nagios' % SLICENAME
+COLLECTD_PID = '/vservers/%s/var/run/collectd.pid' % SLICENAME
+COLLECTD_UNIXSOCK = '/vservers/%s/var/run/collectd-unixsock' % SLICENAME
+LD_LIBRARY_PATH = '/vservers/%s/usr/lib' % SLICENAME
+
+HOSTNAME = socket.gethostname()
+VSYSPATH_ACL = '/vsys/vs_resource_backend.acl'
+VSYSPATH_BACKEND = '/vsys/vs_resource_backend'
+VSYSPATH_SLICE = '/vservers/%s/vsys/vs_resource_backend.in' % SLICENAME
+DEFAULT_TIMEOUT = 60
+
+# Canonical, nagios exit codes.
+STATE_OK = 0
+STATE_WARNING = 1
+STATE_CRITICAL = 2
+STATE_UNKNOWN = 3
+STATUS_MESSAGES = {
+    STATE_OK: 'OKAY',
+    STATE_WARNING: 'WARNING',
+    STATE_CRITICAL: 'CRITICAL',
+    STATE_UNKNOWN: 'UNKNOWN'
+}
+
+
+class Error(Exception):
+  """Base error class for this module."""
+  pass
+
+
+class NagiosStateError(Error):
+  """A generic nagios status error."""
+
+  def __init__(self, message, status_code=STATE_UNKNOWN):
+    self.status_code = status_code
+    Error.__init__(self, message)
+
+
+class CriticalError(Error):
+  """A base class for all critical errors."""
+  pass
```

Comment:
Still seems a little strange to me, but from my brief search around, I think you're right in that Python isn't so big on baking in the error message the way I was talking about